### DISCUSSION TITLE
Questions about LearnCard

### DISCUSSION THREAD

**Discussion by geepara** [learningeconomy/LearnCard#234] [Help]
Hello! My name is George Paragioudakis and I am CTO & Co-Founder @ Youni, an early-stage EdTech startup. Decentralized credentials are a crucial component that we want to integrate into our. Currently, we mint NFTs upon completion of courses on our site. The NFTs contain metadata relating to the skills learned in the course. We like this solution because it potentially allows users and course creators to unlock more value with their tokens through 3rd party apps/integrations that accept them.

We are debating whether or not we want to start issuing credentials through LearnCard rather than issuing tokens. What would it take for another app (developed independently of Youni) to accept our verified credentials? I've read that credentials can be linked to a web3 wallet. Can they be linked to tokens as well?

**Comment by Custard7** [comment]
Hey @geepara,

Thanks for the question - interoperable independent verification of credentials is the entire reason to have verifiable credentials in the first place!

At its core, you can use LearnCard SDK to verify a credential is valid and untampered:

https://docs.learncard.com/learn-card-sdk/learncard-core/quick-start/accept-and-verify-credentials

Then, validate any additional business logic on top of that, such as verifying that the issuer is a particular DID, or that the credential has a certain payload or type.

This can be done by anyone or any system, and it does not require communicating with the issuer to verify the credential.

If you elaborate more on your use case, I could provide more detailed suggestions 😎

Or, I would be happy to schedule a call with you if you want to chat through your use case more. Just shoot an email at sdk@learningeconomy.io and ask to setup a time with me!

**Comment by Custard7** [comment]
Also, terms of your second question, yes!

Credentials can be linked to a web3 wallet, and they can also be linked to tokens.

How they are 'linked' depends on your use case and goals.

One way a wallet can be linked, is by deriving a DID from the key material of a wallet, such as MetaMask. This can then be used as the issuer DID, or subject DID. 

A great example is to use did:pkh and/or did:eth:

https://github.com/w3c-ccg/did-pkh

did:pkh doesn't require posting a did doc on chain, it can be purely generative like did:key, but it can be upgraded to store on chain did document.

In terms of token association:

Again it kind of depends on your use case, but you could link the token to a credential via dids, or if you want a token to have a more complex association, such as an nft issuing a credential, you could potentially use something like the lit protocol:

https://litprotocol.com/

What are you trying to achieve with the token association?

**Comment by Custard7** [comment]
We have a metamask snap that shows the derivation of a DID from a metamask wallet if you look in /services directory as well

**Comment by Developer2Assuredocs** [comment]
Hi team,

I'm encountering an issue with the @learncard/init package. After installing it via pnpm, the expected dist/ directory is missing, which causes module resolution to fail when running the app.

Error: Cannot find module '@learncard/init/dist/index.js'
The package’s package.json contains the following exports:
"exports": {
  ".": {
    "import": "./dist/init.esm.js",
    "require": "./dist/index.js"
  }
}
However, the dist/ directory is not present in the installed package.

It appears the build step may not have been executed prior to publishing, or dist/ is missing from the files list or incorrectly excluded via .npmignore.

Request:

Could you please verify if:

The package was built (pnpm build or equivalent) before publishing?

The dist/ directory is correctly included in the final package?

Let me know if you’d like me to test a patch or if a republished version is planned. 
Thanks!
